tests/integration/devfile/cmd_devfile_delete_test.go

```diff
@@ -31,187 +38,250 @@ var _ = Describe("odo devfile delete command tests", func() {
+		Context("invalid delete commands", func() {
+			It("should throw an error when only --project flag is passed", func() {
+				helper.CmdShouldFail("odo", "delete", "--project", commonVar.Project, "-f")
+			})
+			It("should throw an error with component name and --all flag", func() {
+				helper.CmdShouldFail("odo", "delete", componentName, "--all", "-f")
+			})
+			It("should throw an error with component name, --app, --project and --all flag", func() {
+				helper.CmdShouldFail("odo", "delete", componentName, "--app", appName, "--project", commonVar.Project, "--all", "-f")
+			})
+			It("should throw an error with component name and --context flag", func() {
+				helper.CmdShouldFail("odo", "delete", componentName, "--context", commonVar.Context, "-f")
+			})
+			It("should throw an error with --project and --context flag", func() {
+				helper.CmdShouldFail("odo", "delete", "--project", commonVar.Project, "--context", commonVar.Context, "-f")
+			})
 		})
 
-		It("should execute preStop events if present", func() {
-			helper.CmdShouldPass("odo", "create", "nodejs", "--project", commonVar.Project, componentName)
-			helper.CopyExample(filepath.Join("source", "devfiles", "nodejs", "project"), commonVar.Context)
-			helper.CopyExampleDevFile(filepath.Join("source", "devfiles", "nodejs", "devfile-with-valid-events.yaml"), filepath.Join(commonVar.Context, "devfile.yaml"))
+		When("the component has resources attached to it", func() {
+			resourceTypes := []string{helper.ResourceTypeDeployment, helper.ResourceTypePod, helper.ResourceTypeService, helper.ResourceTypeIngress, helper.ResourceTypePVC}
 
-			helper.CmdShouldPass("odo", "push", "--project", commonVar.Project)
+			JustBeforeEach(func() {
+				helper.CmdShouldPass("odo", "url", "create", "example", "--host", "1.2.3.4.nip.io", "--port", "3000", "--ingress")
 
-			output := helper.CmdShouldPass("odo", "delete", "--project", commonVar.Project, "-f")
-			helper.MatchAllInOutput(output, []string{
-				fmt.Sprintf("Executing preStop event commands for component %s", componentName),
-				"Executing myprestop command",
-				"Executing secondprestop command",
-				"Executing thirdprestop command",
+				if os.Getenv("KUBERNETES") != "true" {
+					helper.CmdShouldPass("odo", "url", "create", "example-1", "--port", "3000")
+					resourceTypes = append(resourceTypes, helper.ResourceTypeRoute)
+				}
+				helper.CmdShouldPass("odo", "storage", "create", "storage-1", "--size", "1Gi", "--path", "/data1", "--context", commonVar.Context)
+			})
+			JustAfterEach(func() {})
+			When("the component is pushed", func() {
+				JustBeforeEach(func() {
+					helper.CmdShouldPass("odo", "push", "--project", commonVar.Project)
+				})
+				It("should delete the component and its owned resources", func() {
+					helper.CmdShouldPass("odo", "delete", "-f")
+
+					for _, resourceType := range resourceTypes {
+						commonVar.CliRunner.WaitAndCheckForExistence(resourceType, commonVar.Project, 1)
```

Comment:
Isn't this test flaky? It seems to me some resources can be deleted at this moment, especially de deployment

Reply:
Can or cannot?

Reply:
This has worked fine for me every time I have run it.

Reply:
OK, so let's keep it as is